lang/py3/avro/schema.py

```diff
@@ -166,10 +166,11 @@ def __init__(self, items=None, **kwargs):
     else:
       super(ImmutableDict, self).__init__(**kwargs)
 
-  def __setitem__(self, key, value):
-    raise Exception(
-        'Attempting to map key %r to value %r in ImmutableDict %r'
-        % (key, value, self))
+# Zubair: OrderedDict internally invokes this in its constructor so no can do!
+#   def __setitem__(self, key, value):
```

Comment:
Does it mean that we can't ensure immutability on `__setitem__` calls?
If I am following correctly, this is more like a conservative check, so, we should be fine without it.